sdk/src/dlob/DLOB.ts

```diff
@@ -779,9 +779,21 @@ export class DLOB {
 		const bidOrder = bidNode.order;
 		const askOrder = askNode.order;
 
+		const bidUserAuthority = this.userMap.getUserAuthority(
+			bidNode.userAccount.toString()
+		);
+		const askUserAuthority = this.userMap.getUserAuthority(
+			askNode.userAccount.toString()
+		);
+
 		// Can't match two maker orders or if maker and taker are the same
+		const sameAuthority = bidUserAuthority.equals(askUserAuthority);
 		const makerIsTaker = bidNode.userAccount.equals(askNode.userAccount);
```

Comment:
we can get rid of makerIsTaker since the authority check is a superset of the functionality